jupyter_kernel_mgmt/subproc/launcher.py

```diff
@@ -7,23 +7,45 @@
 import json
 import os
 import re
-import six
 import socket
 import stat
-from subprocess import PIPE
+from subprocess import PIPE, Popen
 import sys
 from traitlets.log import get_logger as get_app_logger
 import warnings
 
-from ipython_genutils.encoding import getdefaultencoding
 from jupyter_core.paths import jupyter_runtime_dir, secure_write
 from jupyter_core.utils import ensure_dir_exists
 from ..localinterfaces import localhost, is_local_ip, local_ips
-from .manager import KernelManager
+from .manager import KernelManager, use_sync_subprocess
```

Comment:
Unfortunately it would be possible to monkeypatch `jupyter_kernel_mgmt.subproc.launcher.use_sync_subprocess = True` and not affect the variable in the manager module. You could also have a separate launcher implementation using the same manager.

I think the `use_sync_subprocess` should be defined in this module, where it decides which process wrapper to launch, and then the manager class should do an `isinstance()` check to find out whether it's got a Popen or an async Process.

Reply:
Good idea, thanks.